### PR TITLE
Add logic to enable/disable looking for secret in cross namesapce

### PR DESCRIPTION
**Pull request by Bhagyashreek8**

### CODE REVIEW

File: cmd/provisioner/main.go [hunk 1]
```diff
@@ -103,6 +103,12 @@ func main() {
 		"set 'true' to configure bucket quota limit",
 	)
 
+	s3fsprovisioner.AllowSecretNs = flag.Bool(
+		"allowSecretNs",
+		true,
+		"set 'false' to disable allowing to set secret-namespace different from pvc namespace",
```
[the remark this answers is not in the document]
Comment (Bhagyashreek8) [reply cmd/provisioner/main.go:109]: "set to 'false' to disable COS secret lookup in namespace other than PVC's namespace"

File: provisioner/ibm-s3fs-provisioner.go [hunk 2]
```diff
@@ -246,11 +247,18 @@ func (p *IBMS3fsProvisioner) validateAnnotations(ctx context.Context, options co
...
-	if pvc.SecretNamespace == "" {
-		if sc.SecretNamespace != "" {
-			pvc.SecretNamespace = sc.SecretNamespace
-		} else {
-			pvc.SecretNamespace = options.PVC.Namespace
+	contextLogger.Info(pvcName + ":" + clusterID + " AllowSecretNs: " + strconv.FormatBool(*AllowSecretNs))
+
+	if AllowSecretNs != nil && *AllowSecretNs == false {
+		contextLogger.Info(pvcName + ":" + clusterID + " AllowSecretNs is set to false, the secret will be looked for in same namespace where pvc is created")
+		pvc.SecretNamespace = options.PVC.Namespace
```
Comment (Bhagyashreek8) [inline provisioner/ibm-s3fs-provisioner.go:254]: if (pvc.SecretNamespace != "") -> add warning saying "ibm.io/secret-namespace annotation even if set will be ignored"

**Review by nkkashyap — APPROVED** [review]
LGTM

File: provisioner/ibm-s3fs-provisioner.go [hunk 3]
```diff
@@ -246,11 +247,21 @@ func (p *IBMS3fsProvisioner) validateAnnotations(ctx context.Context, options co
...
-		if sc.SecretNamespace != "" {
-			pvc.SecretNamespace = sc.SecretNamespace
-		} else {
-			pvc.SecretNamespace = options.PVC.Namespace
+	contextLogger.Info(pvcName + ":" + clusterID + " AllowCrossNsSecret: " + strconv.FormatBool(*AllowCrossNsSecret))
+
+	if AllowCrossNsSecret != nil && *AllowCrossNsSecret == false {
+		contextLogger.Info(pvcName + ":" + clusterID + " AllowCrossNsSecret is set to false, the secret will be looked for in same namespace where pvc is created")
+		if pvc.SecretNamespace != "" {
+			contextLogger.Warn(pvcName + ":" + clusterID + " ibm.io/secret-namespace annotation if set will be ignored as AllowCrossNsSecret is set to false")
```
Comment (mssachan) [inline provisioner/ibm-s3fs-provisioner.go:255]: `ibm.io/secret-namespace annotation if set will be ignored as AllowCrossNsSecret is set to false` ---> `Ignoring 'ibm.io/secret-namespace' annotation as AllowCrossNsSecret is set to false`

File: provisioner/ibm-s3fs-provisioner_test.go [hunk 4]
```diff
@@ -1075,9 +1077,52 @@ func Test_Provision_ConfigBucketAccessPolicy_ExistingBucket_AutoCreateBucket(t *
...
+	allowCrossNsSect := true
+	AllowCrossNsSecret = &allowCrossNsSect
+	v.PVC.Namespace = "pvc-namespace"
+	v.PVC.Annotations[annotationSecretNamespace] = testNamespace
+	pv, _, err := p.Provision(context.Background(), v)
+	assert.NoError(t, err)
+	assert.Equal(t, testNamespace, pv.Annotations[annotationSecretNamespace])
+}
+
+func Test_Provision_AllowCrossNsSecret_False_SetDiffSecretNS(t *testing.T) {
```
Comment (mssachan) [inline provisioner/ibm-s3fs-provisioner_test.go:1095]: `Test_Provision_AllowCrossNsSecret_False_SetDiffSecretNS` ---> `Test_Provision_AllowCrossNsSecret_False_SetDiffSecretNS_Negative`

**Pull request merged**